packaging/resource_suite_s3_cache.py

```diff
@@ -608,3 +635,237 @@ def test_multiple_s3_endpoints_replication_issue1858(self):
+            s3_client.make_bucket(self.s3bucketname, location=self.s3region)
+
+        # set up resources
+
+        hostname = lib.get_hostname()
+        s3params = 'S3_RETRY_COUNT=15;S3_WAIT_TIME_SECONDS=1;S3_PROTO=%s;S3_MPU_CHUNK=10;S3_MPU_THREADS=4;S3_ENABLE_MD5=1' % self.proto
+        s3params += ';S3_STSDATE=' + self.s3stsdate
+        s3params += ';S3_DEFAULT_HOSTNAME=' + self.s3endPoint
+        s3params += ';S3_AUTH_FILE=' +  self.keypairfile
+        s3params += ';S3_REGIONNAME=' + self.s3region
+        s3params += ';ARCHIVE_NAMING_POLICY=' + self.archive_naming_policy
+        if hasattr(self, 's3sse'):
+            s3params += ';S3_SERVER_ENCRYPT=' + str(self.s3sse)
+
+        s3params=os.environ.get('S3PARAMS', s3params);
+
+        with session.make_session_for_existing_admin() as admin_session:
+            irods_config = IrodsConfig()
+            admin_session.assert_icommand("iadmin modresc demoResc name origResc", 'STDOUT_SINGLELINE', 'rename', input='yes\n')
+            admin_session.assert_icommand("iadmin mkresc demoResc compound", 'STDOUT_SINGLELINE', 'compound')
+            admin_session.assert_icommand("iadmin mkresc cacheResc 'unixfilesystem' " + hostname + ":" + irods_config.irods_directory + "/cacheRescVault", 'STDOUT_SINGLELINE', 'cacheResc')
+            admin_session.assert_icommand('iadmin mkresc archiveResc s3 '+hostname+':/'+self.s3bucketname+'/irods/Vault "'+s3params+'"', 'STDOUT_SINGLELINE', 'archiveResc')
+            admin_session.assert_icommand("iadmin addchildtoresc demoResc cacheResc cache")
+            admin_session.assert_icommand("iadmin addchildtoresc demoResc archiveResc archive")
+
+    def tearDown(self):
+        super(Test_S3_Cache_Glacier_Base, self).tearDown()
+
+        # delete s3 bucket
+        try:
+            httpClient = urllib3.poolmanager.ProxyManager(
+                os.environ['http_proxy'],
+                timeout=urllib3.Timeout.DEFAULT_TIMEOUT,
+                cert_reqs='CERT_REQUIRED',
+                retries=urllib3.Retry(
+                    total=5,
+                    backoff_factor=0.2,
+                    status_forcelist=[500, 502, 503, 504]
+                )
+            )
+        except KeyError:
+            httpClient = None
+
+        if self.proto == 'HTTPS':
+            s3_client = Minio(self.s3endPoint,
+                              access_key=self.aws_access_key_id,
+                              secret_key=self.aws_secret_access_key,
+                              http_client=httpClient,
+                              region=self.s3region)
+        else:
+            s3_client = Minio(self.s3endPoint,
+                              access_key=self.aws_access_key_id,
+                              secret_key=self.aws_secret_access_key,
+                              http_client=httpClient,
+                              region=self.s3region,
+                              secure=False)
```

Comment:
These can be collapsed where `secure=(self.proto == 'HTTPS')`.

Reply:
Done for all.